src/Flysystem/FlysystemAssetStore.php

```diff
@@ -1198,9 +1197,10 @@ function (
         }
 
         // Submit and validate result
-        $result = $callback($fs, $parsedFileID->getFileID());
-        if (!$result) {
-            throw new FlysystemException("Could not save {$filename}");
+        try {
+            $callback($fs, $parsedFileID->getFileID());
+        } catch (UnableToWriteFile $exception) {
+            throw $exception;
         }
```

Comment:
```suggestion
        $callback($fs, $parsedFileID->getFileID());
```
If all we're doing is re-throwing the exception after catching it, we may as well not bother catching it

Reply:
DONE